cmd/operator-sdk/generate/bundle/bundle.go

```diff
@@ -0,0 +1,217 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+
+	genutil "github.com/operator-framework/operator-sdk/cmd/operator-sdk/generate/internal"
+	gencsv "github.com/operator-framework/operator-sdk/internal/generate/clusterserviceversion"
+	"github.com/operator-framework/operator-sdk/internal/generate/collector"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/kustomize"
+)
+
+// TODO(estroz): make these paths relative to inputDir
+const kubebuilderKustomization = `resources:
+- ../default
+- ../samples
+`
+
+func (c *bundleCmd) setDefaults(cfg *config.Config) {
+	if c.operatorName == "" {
+		c.operatorName = filepath.Base(cfg.Repo)
+	}
+}
+
+func (c bundleCmd) runKustomize(cfg *config.Config) error {
+
+	if !c.quiet {
+		fmt.Println("Generating bundle manifest kustomize bases")
+	}
+
+	defaultDir := filepath.Join("config", "bundle")
+	if c.inputDir == "" {
+		c.inputDir = defaultDir
+	}
+	if c.outputDir == "" {
+		c.outputDir = defaultDir
+	}
+	if c.apisDir == "" {
+		if cfg.MultiGroup {
+			c.apisDir = "apis"
+		} else {
+			c.apisDir = "api"
+		}
+	}
+
+	csvGen := gencsv.Generator{
+		OperatorName: c.operatorName,
+		OperatorType: genutil.PluginKeyToOperatorType(cfg.Layout),
+	}
+	opts := []gencsv.Option{
+		gencsv.WithBase(c.inputDir, c.apisDir),
+		gencsv.WithBaseWriter(c.outputDir),
+	}
+	if err := csvGen.Generate(cfg, opts...); err != nil {
+		return fmt.Errorf("error generating ClusterServiceVersion: %v", err)
+	}
+
+	if err := kustomize.WriteIfNotExist(c.outputDir, kubebuilderKustomization); err != nil {
+		return err
+	}
+
+	if !c.quiet {
+		fmt.Println("Bases generated successfully")
+	}
+
+	return nil
+}
+
+func (c bundleCmd) validateManifests(*config.Config) (err error) {
+	if c.version != "" {
+		if err := genutil.ValidateVersion(c.version); err != nil {
+			return err
+		}
+	}
+
+	if !genutil.IsPipeReader() {
+		if c.manifestRoot == "" {
+			return errors.New("--manifest-root must be set if not reading from stdin")
+		}
+		if c.crdsDir == "" {
+			return errors.New("--crds-dir must be set if not reading from stdin")
+		}
+	}
+
+	if c.stdout {
+		if c.outputDir != "" {
+			return errors.New("--output-dir cannot be set if writing to stdout")
+		}
+	}
+
+	return nil
+}
+
+func (c bundleCmd) runManifests(cfg *config.Config) (err error) {
```

Comment:
We have been telling for the users use the `make` commands in all docs.
So, why should we port the make manifest and etc for go? Why not just call here the make commands instead of it?